src/mlpack/tests/cv_test.cpp

```diff
@@ -90,4 +98,78 @@ BOOST_AUTO_TEST_CASE(MSEMatResponsesTest)
   BOOST_REQUIRE_CLOSE(MSE::Evaluate(ffn, data, responses), expectedMSE, 1e-1);
 }
 
+template<typename Class,
+         typename ExpectedPT,
+         typename PassedMT = arma::mat,
+         typename PassedPT = arma::Row<size_t>>
+void CheckPredictionsType()
+{
+  using Extractor = MetaInfoExtractor<Class, PassedMT, PassedPT>;
+  using ActualPT = typename Extractor::PredictionsType;
+  static_assert(std::is_same<ExpectedPT, ActualPT>::value,
+      "Should be the same");
+}
+
+BOOST_AUTO_TEST_CASE(PredictionsTypeTest)
+{
+  CheckPredictionsType<LinearRegression, arma::rowvec>();
+  CheckPredictionsType<FFN<>, arma::mat>();
+
+  CheckPredictionsType<LogisticRegression<>, arma::Row<size_t>>();
+  CheckPredictionsType<SoftmaxRegression<>, arma::Row<size_t>>();
+  CheckPredictionsType<HoeffdingTree<>, arma::Row<size_t>, arma::mat>();
+  CheckPredictionsType<HoeffdingTree<>, arma::Row<size_t>, arma::imat>();
+  CheckPredictionsType<DecisionTree<>, arma::Row<size_t>, arma::mat,
+      arma::Row<size_t>>();
+  CheckPredictionsType<DecisionTree<>, arma::Row<char>, arma::mat,
+      arma::Row<char>>();
```

Comment:
Just out of curiosity, did you try alternate types for other classes than `DecisionTree<>` and have a failure?  I think since #290 is not done yet, there may be problems with that for some classes.